### DISCUSSION TITLE
progress and output from `docker build`?

### DISCUSSION THREAD

**Discussion by mikeblas** [testcontainers/testcontainers-dotnet#973] [Q&A] [answered]
Some of my images take a while to build, particularly if the cache is empty/cold. If I do an actual `docker build` command line, they generate output while building and that's comforting.

Is it possible to catch that output when using TestContainers? I don't see a way to do it -- seems like the only result from the build is an exception (or not). It would be great if I could catch a stream from `docker build --progress=plain`, but I just don't see a way to do it.

**Comment by HofmeisterAn** [comment, marked as answer]
The messages for building and pulling images are logged using the debug log level. You can activate debug log messages in the default logger setup by enabling `ConsoleLogger.Instance.DebugLogLevelEnabled`, or you can use your own custom logger by setting `TestcontainersSettings.Logger` that forwards the messages.

**Reply by mikeblas** [reply]
I wish I could report that worked.  Logging to debug level in my logger seems to work fine, but during a long image build, I see no messages.

**Reply by mikeblas** [reply]
I ended up discovering this was supported in a newer version of the library. After upgrading, the debug log does include the image build output. I'd still rather have a listener (like the output consumers) than plumb this through the logging infrastructure.
